cudf/dataframe.py

```diff
@@ -692,6 +693,64 @@ def _n_largest_or_smallest(self, method, n, columns, keep):
                 df[k] = self[k].reset_index().take(new_positions)
         return df.set_index(self.index.take(new_positions))
 
+    def transpose(self):
+        """Transpose index and columns.
+
+        Returns
+        -------
+        a new (ncol x nrow) dataframe. self is (nrow x ncol)
+
+        Difference from pandas
+        ----------------------
+        Not supporting *copy* because default and only behaviour is copy=True
+        """
+        dtype = self.dtypes[0]
+        if any(t != dtype for t in self.dtypes):
+            raise ValueError('all columns must have the same dtype')
+        has_null = any(c.has_null_mask for c in self._cols.values())
```

Comment:
We should check the `null_count` property instead of the existence of a null mask here in case we decide to always allocate a null mask when all values are valid.